src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -17,35 +21,40 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** 
-* Handles adding the form submission to the datastore in 
-* POST and returns json of the user's profile information
-* to client.
-*/
+/**
+ * Handles adding the form submission to the datastore in
+ * POST and returns json of the user's profile information
+ * to client.
+ */
 @WebServlet("/profile-data")
 public class ProfileServlet extends HttpServlet {
-  
   private static final String NAME_PROPERTY_KEY = "name";
   private static final String EMAIL_PROPERTY_KEY = "email";
   private static final String USERNAME_PROPERTY_KEY = "username";
   private static final String BIO_PROPERTY_KEY = "bio";
+  private static final String EDIT_PROPERTY_KEY = "edit";
+
   ProfileDatastore profileData = new ProfileDatastore();
+  String editValue;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-
     String name = request.getParameter(NAME_PROPERTY_KEY);
     String email = request.getParameter(EMAIL_PROPERTY_KEY);
-    String username = request.getParameter(USERNAME_PROPERTY_KEY );
+    String username = request.getParameter(USERNAME_PROPERTY_KEY);
     String bio = request.getParameter(BIO_PROPERTY_KEY);
+    editValue = request.getParameter(EDIT_PROPERTY_KEY);
```

Comment:
Boolean.parseBoolean() takes a string argument and will return `true` if that argument is not null and equals "true"

```suggestion
    boolean edit = Boolean.parseBoolean(request.getParameter(EDIT_PROPERTY_KEY);
```